src/internal/utils.ts

```diff
@@ -35,3 +35,9 @@ export function debounce<T extends Function>(func: T, milliseconds: number, imme
     return result;
   };
 };
+
+export function isES6Map(thing: any) {
+  if (Map !== undefined && thing instanceof Map)
```

Comment:
```suggestion
  if (typeof Map !== 'undefined' && thing instanceof Map)
```

https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html (check for root level undefined) :rose: